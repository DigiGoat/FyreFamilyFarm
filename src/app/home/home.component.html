<div class="w-100 text-center">
  <img src="./assets/icons/android-chrome-192x192.png" width="150" class="text-center">
</div>
<h1 class="text-center">{{homeTitle}}</h1>
<h5 class="text-center">{{owner}}</h5>
<h6 class="text-center"><a [href]="'mailto:' + email">{{email}}</a></h6>
<!--<h2 class="text-center">Bloom Family Farm Nigerians</h2>-->
<div class="container">
  <p class="text-center" [innerHTML]="homeDescription"></p>
</div>
