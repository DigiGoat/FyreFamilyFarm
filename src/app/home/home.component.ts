import { Component } from '@angular/core';

import app from '../../assets/resources/app.json';
import { AppService } from '../services/app.service';


@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss']
})
export class HomeComponent {
  constructor(public app: AppService) { }
}
