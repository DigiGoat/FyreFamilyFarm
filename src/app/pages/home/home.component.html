<div class="w-100 text-center">
  <img ngSrc="./assets/icons/android-chrome-192x192.png" alt="Farm Logo" width="150" height="150" class="text-center" priority>
</div>
<h1 class="text-center">{{config.homeTitle}}</h1>
<h5 class="text-center">{{config.owner}}</h5>
<h6 class="text-center"><a [href]="'mailto:' + config.email">{{config.email}}</a></h6>
<div class="container">
  <p class="text-center" [innerHTML]="config.homeDescription"></p>
</div>
