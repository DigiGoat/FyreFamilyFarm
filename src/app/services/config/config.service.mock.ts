import type { ConfigService } from './config.service';

export const ConfigServiceMock = {
  email: 'TEST_EMAIL',
  homeDescription: 'TEST_DESCRIPTION',
  homeTitle: 'TEST_TITLE',
  menubarTitle: 'TEST_MENUBAR_TITLE',
  owner: 'TEST_OWNER',
  repo: 'TEST_REPO'
} as ConfigService;
