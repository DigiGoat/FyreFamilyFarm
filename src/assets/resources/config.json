{
  "owner": "Angel Fyre",
  "email": "fyrefamfarm@gmail.com",
  "homeTitle": "Fyre Family Farm",
  "menubarTitle": "Fyre Family Farm",
  "tabTitle": "Fyre Family"
}
