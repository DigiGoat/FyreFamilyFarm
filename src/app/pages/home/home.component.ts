import { Component, type OnInit } from '@angular/core';

import type { Observable } from 'rxjs';
import type { Page } from '../../app-routing.module';
import { ConfigService } from '../../services/config/config.service';
import { Meta } from '@angular/platform-browser';


@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss']
})
export class HomeComponent implements Page, OnInit {
  public description = '';
  constructor(public config: ConfigService, private meta: Meta) { }
  ngOnInit(): void {
    this.description = this.config.homeDescription;
    this.setDescription();
  }
  setDescription(): void | Observable<void> {
    if (this.config.homeDescription) {
      this.meta.addTags([{ name: 'og:description', content: this.description }, { name: 'description', content: this.description }]);
    }
  }
}
