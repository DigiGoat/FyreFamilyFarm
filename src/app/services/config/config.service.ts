import { Injectable } from '@angular/core';

import config from '../../../assets/resources/config.json';


@Injectable({
  providedIn: 'root'
})
export class ConfigService {
  private config: Record<string, any> = config;
  get homeTitle(): string {
    if (this.config['homeTitle']) {
      return this.config['homeTitle'];
    } else {
      return 'Dairy Goat Farm';
    }
  };
  get owner(): string {
    if (this.config['owner']) {
      return this.config['owner'];
    } else {
      return 'First & Last Name';
    }
  }
  get email(): string {
    if (this.config['email']) {
      return this.config['email'];
    } else {
      return 'someone@email.com';
    }
  }
  get homeDescription(): string {
    if (this.config['homeDescription']) {
      return this.config['homeDescription'];
    } else {
      return 'Our Farm is home to ...';
    }
  }
  get repo(): string {
    if (this.config['repo']) {
      return this.config['repo'];
    } else {
      return 'https://github.com/DigiGoat/your-website';
    }
  }
  get menubarTitle(): string {
    if (this.config['menubarTitle']) {
      return this.config['menubarTitle'];
    } else {
      return 'Dairy Goat <br> Farm';
    }
  }
  get tabTitle(): string {
    if (this.config['tabTitle']) {
      return this.config['tabTitle'];
    } else {
      return 'Dairy Goat Farm';
    }
  }
}
