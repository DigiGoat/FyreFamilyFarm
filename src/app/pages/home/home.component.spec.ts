import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ConfigService } from '../../services/config/config.service';
import { ConfigServiceMock } from '../../services/config/config.service.mock';
import { HomeComponent } from './home.component';


jest.mock('../../services/config/config.service');
const configService = jest.mocked(ConfigService);
describe('HomeComponent', () => {
  let component: HomeComponent;
  let fixture: ComponentFixture<HomeComponent>;


  beforeEach(async () => {
    configService.mockClear();
    await TestBed.configureTestingModule({
      declarations: [HomeComponent],
      providers: [ConfigService],
    })
      .compileComponents();
    fixture = TestBed.createComponent(HomeComponent);
    TestBed.inject(ConfigService);
    component = fixture.componentInstance;

    configService.mockImplementation(() => {
      return ConfigServiceMock;
    });
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should call ConfigService once', () => {
    expect(configService).toHaveBeenCalledTimes(1);
  });
  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
