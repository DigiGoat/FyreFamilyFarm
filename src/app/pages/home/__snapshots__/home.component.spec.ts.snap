// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render 1`] = `
<app-home
  config={[Function Object]}
>
  <div
    class="w-100 text-center"
  >
    <img
      class="text-center"
      height="150"
      ngsrc="./assets/icons/android-chrome-192x192.png"
      priority=""
      width="150"
    />
  </div><h1
    class="text-center"
  >
    TEST_TITLE
  </h1><h5
    class="text-center"
  >
    TEST_OWNER
  </h5><h6
    class="text-center"
  >
    <a
      href="mailto:TEST_EMAIL"
    >
      TEST_EMAIL
    </a>
  </h6><div
    class="container"
  >
    <p
      class="text-center"
    >
      TEST_DESCRIPTION
    </p>
  </div>
</app-home>
`;
